instrumentation/elasticsearch/lib/opentelemetry/instrumentation/elasticsearch/patches/client.rb

```diff
@@ -0,0 +1,62 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+module OpenTelemetry
+  module Instrumentation
+    module Elasticsearch
+      module Patches
+        NAME_FORMAT = '%s %s'
+        IP_TCP = 'ip_tcp'
+
+        # Module to prepend to Elastic::Transport::Client for instrumentation
+        module Client
+          # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+          def perform_request(method, path, *args, &block)
+            config = Elasticsearch::Instrumentation.instance.config
+            attributes = {
+              'db.system' => 'elasticsearch',
+              # TODO: should we set db.name?
+              # 'db.name' => database_name,
+              'db.operation' => method,
+              # TODO: is this true for all elasticsearch client requests?
+              'net.transport' => IP_TCP,
```

Comment:
that field isn't that interesting here given the http client itself might be instrumented?
and if that isn't the case than shouldn't the http.* attributes be of more interest here?

Reply:
I didn't see that there was an PR already for this instrumentation, thanks for pointing that out. I'll take a look at it and see how these can be merged.
When you say that the approach is one you would recommend, you mean that there's something about @plantfansam's approach that is preferable to the one in this PR? If so, I might have some follow-up questions to clarify what exactly you'd prefer to have in the instrumentation as I take a look.

Reply:
Update: I've taken a look at the other PR and I think it makes a lot of sense to use `OpenTelemetry::Common::HTTP::ClientContext.with_attributes`. We also have a way of avoiding redundant spans in the Elastic APM agent's Elasticsearch client instrumentation. So I'll add that in, including the config option, so the user has a choice.

Is there anything else about the approach in the other PR that you'd like to include in this one?

Reply:
Update: I've added a config option `capture_es_spans` to the instrumentation so that the user can decide whether they want the http span to be enriched with elasticsearch instrumentation attributes or if they want separate spans for es and http.